ably/paginated_result.go

```diff
@@ -80,11 +74,10 @@ func newPaginatedResult(opts *proto.ChannelOptions, typ reflect.Type, path strin
 			}
 			rs[k] = m
 		}
-		p.typItems = rs
-		return p, nil
+		return rs, nil
 	case presMsgType:
```

Comment:
I don't really understand why there is special-case code in here to handle `Message` and `PresenceMessage` whereas other types (eg `Stats`) are handled differently

Reply:
Because `Message` and `PresenceMessage`  requires special handling when decoding.

```go
		var o []map[string]interface{}
		err := decodeResp(resp, &o)
		if err != nil {
			return nil, err
		}
		rs := make([]*proto.Message, len(o))
		for k, v := range o {
			m := &proto.Message{
				ChannelOptions: opts,
			}
			err = m.FromMap(v)
			if err != nil {
				return nil, err
			}
			rs[k] = m
		}
		return rs, nil
```

see, we also have to pass the `ChannelOptions` object in case the message will need decryption. Same applies to `PresenceMessage`.  For other values we fallback to default `json.Unmarshal`.

This is implementation detail so I don't think it will affect the user's API. 
I'm happy to hear other suggestion on a better way to handle this situation.